woocommerce/google-listings-and-ads
# Use Ads' currency inside Dashboard's table.

### Changes proposed in this Pull Request:

- Make `/wc/gla/ads/connection` return also currency symbol. (187d539)
	It is easier to use the code to symbol dictionary directly on the server-side. And we will need it in the majority of cases when we check the Ads currency.
- Add `useAdsCurrency` hook, use it in the Dashboard table (a4f01a5)
	Keep using currency format as specified in the shop's settings, but fetch and display Ads' account currency and symbol, to display the correct budget values.
	Solves part of https://github.com/woocommerce/google-listings-and-ads/issues/363#issuecomment-883081178https://github.com/woocommerce/google-listings-and-ads/issues/363#issuecomment-883081178


### Screenshots:

#### Woo currency settings
![image](https://user-images.githubusercontent.com/17435/127513949-13764896-c1ac-43ae-ba7e-1dfccee551bf.png)

#### Ads currency 
![image](https://user-images.githubusercontent.com/17435/126819728-19a08916-69e1-497d-b189-28cd28aa2bb0.png)

#### Dashboard table currency
![image](https://user-images.githubusercontent.com/17435/127513819-45aefe81-9adc-4a99-8115-5850a52473b6.png)



### Detailed test instructions:

1. [Set the store currency](https://gla1.test/wp-admin/admin.php?page=wc-settings) to _X_ (PLN)
2. Setup Ads account
3. Change the store currency to _Y_ (EUR)
4. Go to the Dashboard ([`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard) )
5. You should see the "Daily budget" column using store's formatting settings, but display Ads' currency symbol.

### Changelog entry

> Fix - Render Ads Account's currency in Dashboard's table



### Additional notes:
1. :scroll: There are some concerns about an additional sequential Google Account request as mentioned in the previous PR https://github.com/woocommerce/google-listings-and-ads/pull/901#issuecomment-887704588. We could consider removing that need in a separate PR.
